### DISCUSSION TITLE
404 when accessing main container via http

### DISCUSSION THREAD

**Discussion by BryanDollery** [openkruise/kruise#922] [Q&A]
I'm using openkruise to build an adaptor sidecar that handles events on a kafka topic by calling a rest API on the main container. However, I'm getting a 404 when making the HTTP call. Here's the code from my sidecar that makes the call:

```javascript
const results = got.put("http://localhost", { json: { event: { ...data }, entity } }).json();
```

I'm not sure that this is the fault of openkruise, but I thought I'd ask if you guys had seen any similar behaviour?

**Comment by FillZpp** [comment]
Emm... I don't think it is fault of Kruise, for sidecarset only helps you inject the sidecar container into pod. Have you ever tried to define the sidecar directly in the app workload and see if the error is still there?

**Reply by BryanDollery** [reply]
I haven't actually, but I'll give it a go now. Like I said, I don't think it's a kruise issue either, but I was just wondering if you'd seen anything like it before (or even if I was missing a kruicial bit of config).

**Reply by FillZpp** [reply]
I'm not sure what's the root cause of it. There may be too many reasons for 404. You may just exec in to the container and use `curl` to see if you can connect to the http server manually.
